build.py

```diff
@@ -670,13 +670,16 @@ def onnxruntime_cmake_args(images, library_paths):
     ]
 
     # TRITON_ENABLE_GPU is already set for all backends in backend_cmake_args()
-    # TODO: TPRD-334 TensorRT extension is not currently supported by our manylinux build
-    if FLAGS.enable_gpu and target_platform() != "rhel":
-        cargs.append(
-            cmake_backend_enable(
-                "onnxruntime", "TRITON_ENABLE_ONNXRUNTIME_TENSORRT", True
+    if FLAGS.enable_gpu:
+        # TODO: TPRD-712 TensorRT is not currently supported by our RHEL build for SBSA.
+        if target_platform() != "rhel" or (
+            target_platform() == "rhel" and target_machine() != "aarch64"
```

Comment:
Since there are only two target machines for `rhel`, I would prefer if this second condition is two positives rather than one positive and one negative. 
```suggestion
            target_platform() == "rhel" and target_machine() == "x86_64"
```